### PR TITLE
add index to connected peer discovery loop

### PR DESCRIPTION
**Pull request by dustinblack**
The xmltodict return on the 'gluster --xml pool list' output results in the 'peers' key being a list, not a dictionary, resulting in a type error backtrace at runtime. This change adds an index value for the list so that it can be properly referenced in the for loop.

### CODE REVIEW

File: zeroconf/actions.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ def connected_peers():
     if not (result['cliOutput'] and result['cliOutput']['peerStatus']):
         return peers
 
-    for peer in result['cliOutput']['peerStatus'].values():
-        hostname = peer['hostname']
+    for idx, peer in enumerate(result['cliOutput']['peerStatus'].values()):
+        hostname = peer[idx]['hostname']
```
Comment (nixpanic) [inline zeroconf/actions.py]: This is unexpected to me. The loop has a `peer` variable that should have all the attributes for itself. Using the `idx` of the list that is returned seems wrong, or at least very unintuitive.

Could you provide an example of the xml output that shows the structure?

Comment (dustinblack) [reply zeroconf/actions.py]: When I run `gluster-discovery` where the peers are already part of a trusted pool, I get the `TypeError` traceback:

```
[root@g1-1 ~]# gluster-discovery 
discovered host: 10.11.12.5
discovered host: 10.11.12.10
discovered host: 10.11.12.9
discovered host: 10.11.12.2
discovered host: 10.11.12.6
discovered host: 10.11.12.7
Traceback (most recent call last):
  File "/usr/bin/gluster-discovery", line 9, in <module>
    load_entry_point('gluster-zeroconf==0.1.0', 'console_scripts', 'gluster-discovery')()
  File "/usr/lib/python2.7/site-packages/gluster/zeroconf/cli.py", line 21, in main
    peers = connected_peers()
  File "/usr/lib/python2.7/site-packages/gluster/zeroconf/actions.py", line 68, in connected_peers
    hostname = peer['hostname']
TypeError: list indices must be integers, not str
```

In actions.py, the XML output of `gluster --xml pool list` is converted to a dict with xmltodict. The `for peer ...` at line 67 then drills down into that that data structure, but if you take a look at the structure of what's mapped to `peer` you find that the top level is a list of ordered dicts.

```
[   OrderedDict([(u'uuid', u'4871010f-54e4-4cba-836c-bc3ae59e3a73'), (u'hostname', u'g1-2'), (u'hostnames', OrderedDict([(u'hostname', u'g1-2')])), (u'connected', u'1'), (u'state', u'3'), (u'stateStr', u'Peer in Cluster')]),
    OrderedDict([(u'uuid', u'a2976756-de17-4190-90c2-514771a0922c'), (u'hostname', u'g1-3'), (u'hostnames', OrderedDict([(u'hostname', u'g1-3')])), (u'connected', u'1'), (u'state', u'3'), (u'stateStr', u'Peer in Cluster')]),
    OrderedDict([(u'uuid', u'39055be9-4352-49a6-afbe-8c37e406a106'), (u'hostname', u'g1-4'), (u'hostnames', OrderedDict([(u'hostname', u'g1-4')])), (u'connected', u'1'), (u'state', u'3'), (u'stateStr', u'Peer in Cluster')]),
    OrderedDict([(u'uuid', u'a9ca0eaa-c07a-47f1-9559-621d5f97a41a'), (u'hostname', u'g1-5'), (u'hostnames', OrderedDict([(u'hostname', u'g1-5')])), (u'connected', u'1'), (u'state', u'3'), (u'stateStr', u'Peer in Cluster')]),
    OrderedDict([(u'uuid', u'0ba2f242-e26b-4cce-903c-591ef93ae58c'), (u'hostname', u'g1-6'), (u'hostnames', OrderedDict([(u'hostname', u'g1-6')])), (u'connected', u'1'), (u'state', u'3'), (u'stateStr', u'Peer in Cluster')]),
    OrderedDict([(u'uuid', u'12eae974-552b-4dc5-9be9-43865856220f'), (u'hostname', u'localhost'), (u'connected', u'1')])]
```
However, I also see now that the patch I provided doesn't completely fix the problem. The backtrace is gone, but it only reports the "connected peer" hostname for one peer.

Comment (dustinblack) [reply zeroconf/actions.py]: OK, I see now that `xmltodict` is collapsing all of the `<peer>` entries in the XML to one `peer` list inside the dictionary. That's odd, and unexpected, but the fix is even a bit simpler.

**Comment by dustinblack** [comment]
Pull requested updated to a commit with a simpler fix for the reported problem -- look into the 'peer' key for a list of entries instead of getting the values() of the 'peerStatus' parent key.

**Pull request merged**